### PR TITLE
CSR v1 - switch controllers

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

* Moves the IsKubelet{Client,Serving}CSR helpers to a version-neutral package
* Switches the kube-controller-manager certificate controllers to the v1 API

xref https://github.com/kubernetes/enhancements/issues/1513

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/milestone v1.19
/priority important-soon
/sig auth
/cc @munnerz @deads2k

### CODE REVIEW

File: pkg/apis/certificates/v1beta1/defaults.go [hunk 1]
```diff
@@ -67,99 +65,24 @@ func DefaultSignerNameFromSpec(obj *certificatesv1beta1.CertificateSigningReques
...
-// the order of the elements
-func equalUnsorted(left, right []certificatesv1beta1.KeyUsage) bool {
-	l := sets.NewString()
-	for _, s := range left {
-		l.Insert(string(s))
-	}
-	r := sets.NewString()
-	for _, s := range right {
-		r.Insert(string(s))
+func usagesToSet(usages []certificatesv1beta1.KeyUsage) sets.String {
```
Comment (munnerz) [inline pkg/apis/certificates/v1beta1/defaults.go:82]: +1 much tidier

File: pkg/controller/certificates/approver/sarapprove.go [hunk 2]
```diff
@@ -146,19 +147,26 @@ func appendApprovalCondition(csr *capi.CertificateSigningRequest, message string
...
 func isSelfNodeClientCert(csr *capi.CertificateSigningRequest, x509cr *x509.CertificateRequest) bool {
-	if !isNodeClientCert(csr, x509cr) {
+	if csr.Spec.Username != x509cr.Subject.CommonName {
 		return false
 	}
-	if csr.Spec.Username != x509cr.Subject.CommonName {
+	if !isNodeClientCert(csr, x509cr) {
 		return false
 	}
 	return true
```
Comment (munnerz) [inline pkg/controller/certificates/approver/sarapprove.go:163]: nit: maybe just `return isNodeClientCert(csr, x509cr)` here?

File: pkg/controller/certificates/signer/signer.go [hunk 3]
```diff
@@ -73,7 +74,7 @@ func NewLegacyUnknownCSRSigningController(
 	caFile, caKeyFile string,
 	certTTL time.Duration,
 ) (*CSRSigningController, error) {
-	return NewCSRSigningController("csrsigning-legacy-unknown", capi.LegacyUnknownSignerName, client, csrInformer, caFile, caKeyFile, certTTL)
+	return NewCSRSigningController("csrsigning-legacy-unknown", "kubernetes.io/legacy-unknown", client, csrInformer, caFile, caKeyFile, certTTL)
```
Comment (munnerz) [inline pkg/controller/certificates/signer/signer.go:77]: not a big issue, but why no longer reference the const here given it does still exist in v1beta1? Reducing the number of things we import/avoiding importing both v1beta1 and v1?

File: pkg/controller/certificates/approver/sarapprove.go (hunk 2 above)
Comment (liggitt) [reply pkg/controller/certificates/approver/sarapprove.go]: done

File: pkg/controller/certificates/signer/signer.go (hunk 3 above)
Comment (liggitt) [reply pkg/controller/certificates/signer/signer.go]: yeah... I was on the fence... switched back to referencing the v1beta1 constant

**Review by liggitt — COMMENTED** [review]
rebased on latest csr-v1 PR, addressed comments

**Pull request merged**